src/jit/lsra.cpp

```diff
@@ -10355,6 +10376,23 @@ void TreeNodeInfo::Initialize(LinearScan* lsra, GenTree* node, LsraLocation loca
     else
     {
         dstCandidates = genRegMask(node->gtRegNum);
+#ifdef ARM_SOFTFP
+        regNumber reg2 = REG_NA;
+        if (node->OperGet() == GT_PUTARG_REG)
```

Comment:
If you don't include that node because you don't expect it, then perhaps in the else clause you could assert that it is not a multireg node.